src/engine/Type.v3

```diff
@@ -418,12 +418,14 @@ class ArrayDecl extends HeapTypeDecl {
 
 // Continuation type declaration. (ext:stack-switching)
 class ContDecl extends HeapTypeDecl {
-	def sig: SigDecl;
+	var sig_ref: HeapType;
+	var sig: SigDecl;
 
-	new(final: bool, supertypes: Array<HeapType>, sig) super(final, supertypes) {}
+	new(final: bool, supertypes: Array<HeapType>, sig_ref) super(final, supertypes) {}
 
 	def render(buf: StringBuilder) -> StringBuilder {
-		return putUid(buf.put2("cont #%d (%q)", heaptype_index, sig.render));
+		if (sig != null) return putUid(buf.put2("cont #%d (%q)", heaptype_index, sig.render));
```

Comment:
You can simplify this to `putUid(buf.put2("cont #%d (%q)", heaptype_index, if(sig != null, sig, sig_ref).render))`